lib/kubeclient/common.rb

```diff
@@ -38,6 +39,7 @@ def initialize_client(
       @ssl_options = ssl_options
       @auth_options = auth_options
       @socket_options = socket_options
+      @http_proxy_uri = (http_proxy_uri.is_a? URI) ? http_proxy_uri.to_s : http_proxy_uri
```

Comment:
@enoodle always calling ```to_s``` should have the same result. I'm in favor of that since it removes handling of a special case